pco/src/bit_reader.rs

```diff
@@ -47,23 +57,52 @@ pub unsafe fn read_uint_at<U: ReadWriteUint, const MAX_U64S: usize>(
   //    For the 3rd u64 and onward, we skip 8 bytes forward. Due to how we
   //    handled the 2nd u64, the most we'll ever need to shift by is
   //    U::BITS - 8, which is safe.
-  let mut res = U::from_u64(u64_at(src, byte_idx) >> bits_past_byte);
-  let mut processed = min(n, 56 - bits_past_byte);
-  byte_idx += 7;
 
-  for _ in 0..MAX_U64S - 1 {
-    res |= U::from_u64(u64_at(src, byte_idx)) << processed;
-    processed += 64;
-    byte_idx += 8;
-  }
+  let res = match MAX_BYTES {
+    4 => {
```

Comment:
Could you use a match guard to flatten this logic? https://doc.rust-lang.org/rust-by-example/flow_control/match/guard.html